src/wix/WixToolset.Core.Burn/Bundles/BurnWriter.cs

```diff
@@ -138,10 +146,19 @@ public bool AppendContainer(Stream containerStream, long containerSize, BurnComm
                     break;
 
                 case Container.Attached:
-                    burnSectionCount = 2;
-                    burnSectionOffsetSize = BURN_SECTION_OFFSET_ATTACHEDCONTAINERSIZE;
                     // TODO: verify that the size in the section data is 0 or the same size.
-                    this.AttachedContainerSize = (uint)containerSize;
+                    uint nextAddress = this.EngineSize;
+                    foreach (ContainerSlot cntnr in this.AttachedContainers)
+                    {
+                        if (cntnr.Address >= nextAddress)
+                        {
+                            nextAddress = cntnr.Address + cntnr.Size;
+                        }
+                    }
```

Comment:
Recalculating `nextAddress` for every container seems unnecessary. Investigate turning `AttachedContainers` to `UInt32[]` and keeping `nextAddress` as a class variable.